packages/tests/tests/filter/unsubscribe.node.spec.ts

```diff
@@ -107,15 +125,23 @@ describe("Waku Filter V2: Unsubscribe", function () {
 
   it("Unsubscribe topics the node is not subscribed to", async function () {
     // Subscribe to 1 topic and send message
-    await subscription.subscribe([TestDecoder], messageCollector.callback);
+    const subscription = await waku.filter.createSubscription([TestDecoder]);
+    subscription.addEventListener(
+      TestDecoder.contentTopic,
+      messageCollector.filterCallback
+    );
     await waku.lightPush.send(TestEncoder, { payload: utf8ToBytes("M1") });
     expect(await messageCollector.waitForMessages(1)).to.eq(true);
 
     expect(messageCollector.count).to.eq(1);
 
-    // Unsubscribe from topics that the node is not not subscribed to and send again
+    // Unsubscribe from topics that the node is not subscribed to and send again
     await subscription.unsubscribe([]);
-    await subscription.unsubscribe(["/test/2/waku-filter"]);
+    try {
+      await subscription.unsubscribe(["/test/2/waku-filter"]);
+    } catch (error) {
```

Comment:
We should move away from try/catch in favour of return error enum

Reply:
tracked part of #1725